bophono/PhonStateKVP.py

```diff
@@ -44,25 +42,27 @@ def doCombineCurEnd(self, endofword, nrc='', nextvowel=''): # nrc = next root co
         self.phon += self.end
 
 
-    def combineWithException(self, exception):
+    def combineWithException(self, exception, tibetanSyllable):
         syllables = exception.split('|')
         for syl in syllables:
             indexplusminus = syl.find('-')
             if indexplusminus == -1:
                 print("invalid exception syllable: "+syl)
                 continue
-            self.combineWith(syl[:indexplusminus], syl[indexplusminus+1:])
+            self.combineWith(syl[:indexplusminus], syl[indexplusminus+1:], tibetanSyllable)
 
-    def combineWith(self, nextroot, nextend):
+    def combineWith(self, nextroot, nextend, tibetanSyllable):
         nextrootconsonant = nextroot
         nextvowel = ''
         self.doCombineCurEnd(False, nextrootconsonant, nextvowel)
         self.position += 1
         if nextrootconsonant == "-":
             self.phon += ""
+        elif re.search(r'^བ[ོ]?[རསད]?(འི)?$', tibetanSyllable) and self.position == 1:
```

Comment:
can you add a comment here about what it does (and possibly why the `2:` thing didn't work?) ?

Reply:
Actually I realize that I had a kind of false positive.

I thought that by doing this regex this on position 1 I was addressing the case of a "wa" type syllable that is *alone*, but actually I was checking that it worked by testing against བར་ཆད and བར་དུ, which are defined as exceptions (as discussed below). Therefore it worked not because of my tweak but because they were treated as exceptions.

The question now remains: is there a way to treat all the wa cases (wa, we, wö, ...) as wa only if it is the second syllable (taken care of by the `2:` exceptions) *or* if the syllable is alone ? Is there a system in place to target a syllable that is alone, or should it be added ?

If we don't do this then all the lines with an odd number of syllables ending with wa will yield "ba" and need to be corrected by hand like, for instance, twice in the kangki lodrö alone:

> གང་གི་ བློ་གྲོས་ སྒྲིབ་གཉིས་ སྤྲིན་བྲལ་ ཉི་ལྟར་ རྣམ་དག་ རབ་གསལ་བས།།
> གང་དག་ སྲིད་པའི་ བཙོན་རར་ མ་རིག་ མུན་འཐུམས་ སྡུག་བསྔལ་ གྱིས་གཟིར་བའི།།

Reply:
Basically the rule should be, I believe:
- 1 syllable only => wa
- 2 syllables starting with བ => ba (བར་ཆད)
- 2 syllables ending with བ => wa (གསལ་བས)
- odd number of syllables ending with བ => wa (རབ་གསལ་བས)

In the case of 1 syllable and 2/odd syllables ending with wa it should also handle vowels and suffixes.

Reply:
well... I can think of examples where 1 syllable only would be ba (ex: ".. jangchub bar" where "bar" is short for "bartu")... In a word segmented text, what case are you thinking of where བ (or similar) would be alone and pronounced wa ? In a text where syllabes are group two by two, we can imagine cases where ba is alone and pronounced wa but KVP won't use that.

I think the rule for KVP could be if བ is the root of any syllalble after the first it should be "wa". I think it's what the `2:` does so it should be fine... that said, if you want to extend bophono for use cases beyond KVP that's fine too. If we do that let's do it in another issue / pull request maybe?

Reply:
Well I was mainly concerned with lines ending with a བ alone, but since the system if actually treating those cases as 3-syllables words instead of splitting into 2+1, it actually falls within the reach of the `2:` rule.

With that case taken care of I'm happy enough with only the two `2:`rules, and I can remove my funky regex.

Reply:
ok sounds good!